pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -206,6 +206,29 @@ func (p *MellanoxPlugin) Apply() error {
 		return err
 	}
 	if vars.FeatureGate.IsEnabled(consts.MellanoxFirmwareResetFeatureGate) {
+		for _, pciAddress := range pciAddressesToReset {
```

Comment:
should we move this logic to MlxResetFW ? i think its better that the logic is contained there.
(method signature might need to change to get the mellanoxNicsStatus map

Reply:
Moved the new logic to MlxResetFW